src/games/rcll/config.clp

```diff
@@ -37,6 +37,206 @@
   )
 )
 
+(defrule config-sync-config-with-challenges-width
+  "If the challange mode is enabled, synchronize overlapping configs"
+  (confval (path "/llsfrb/challenges/enable") (type BOOL) (value TRUE))
+  ?cv <- (confval (path "/llsfrb/game/field/width") (value ?v))
+  (confval (path "/llsfrb/challenges/field/width") (value ?new-v&:(neq ?v ?new-v)))
+  =>
+  (modify ?cv (value ?new-v))
+)
+
+(defrule config-sync-config-with-challenges-height
+  "If the challange mode is enabled, synchronize overlapping configs"
+  (confval (path "/llsfrb/challenges/enable") (type BOOL) (value TRUE))
+  ?cv <- (confval (path "/llsfrb/game/field/height") (value ?v))
+  (confval (path "/llsfrb/challenges/field/height") (value ?new-v&:(neq ?v ?new-v)))
+  =>
+  (modify ?cv (value ?new-v))
+)
+
+(defrule config-sync-config-with-challenges-mirrored
+  "If the challange mode is enabled, synchronize overlapping configs"
+  (confval (path "/llsfrb/challenges/enable") (type BOOL) (value TRUE))
+  ?cv <- (confval (path "/llsfrb/game/field/mirrored") (value ?v))
+  (confval (path "/llsfrb/challenges/field/mirrored") (value ?new-v&:(neq ?v ?new-v)))
+  =>
+  (modify ?cv (value ?new-v))
+)
+
+; Some of the globals actually directly resemble a config value.
+; In that case synchronize the values so that the actual global
+; variables are always up-to-date
+(defrule config-sync-config-with-global-config-width
+  ?cv  <- (confval (path "/llsfrb/game/field/width") (value ?v))
+  ?cv2 <- (confval (path "/llsfrb/globals/field-width") (value ?v2))
+  (test (neq ?v ?v2))
+  =>
+  (if (< (fact-index ?cv) (fact-index ?cv2)) then
+    (modify ?cv (value ?v2))
+  else
+    (modify ?cv2 (value ?v))
+  )
+)
+
+(defrule config-sync-config-with-global-config-height
+  ?cv  <- (confval (path "/llsfrb/game/field/height") (value ?v))
```

Comment:
why do we have two values?  this seems dangerous in terms of usability

Reply:
i agree this design is bad, it was introduced when challenges were first introduced.
The idea is that challenge configs have their own prefix to make configuration easier, but in hindsight i regret that as it introduces these weird situations and also we are not consistent with it anyways (grasping challenge needs to also alter the game config to load the field).

But I would consider this out of scope for this PR to also rewrite challenge configs.